keps/sig-security/3203-auto-refreshing-official-cve-feed/README.md

```diff
@@ -0,0 +1,487 @@
+
+#### Misuse of Auto-Refresh feature
+
+Without proper filtering and control over who can label GitHub issues, the list
+of CVEs can become a list with poor signal-to-noise ratio making the list
+unusable.
+
+For this purpose, the filtering is applied such that only issues that are marked
+as `closed`
+will be part of the list. Also, additionally, the `official-cve-feed` label is a
+[restricted](https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L140-L150)
+label that can only be applied by SRC and SIG Security Tooling Leads.
+
+#### Large JSON blob could lead to slower read/write and resource consumption
+
+Blobs will only be rewritten, if the generated blob is different from existing
+blob. As hash file would be created and stored alongside generated blob. This
+hash file will be check everytime before push to the hash of the generated file.
+If the hash file matches writing to the bucket will be skipped, if hash file is
+different writing to bucket, will be triggered.
+
+## Design Details
+
+The steps to implement this design will involve a prow job that:
+
+1. Queries Github API for fixed official CVEs
+1. Generates a JSON blob based on the query results
+1. Writes the JSON blob to gcs-bucket if it is different than existing blob
+1. Triggers the `k/website` build using netlify
+   [build-hook](https://docs.netlify.com/configure-builds/build-hooks/). Secret
+   token to trigger build is added as External Secret. See example for
+   [snyk-token](https://github.com/kubernetes/k8s.io/pull/2222/files)
+1. `k/website` build pulls the JSON blob from gcs bucket during website rebuild,
+   pulling it from gcs-bucket into something like
+   `https://kubernetes.io/security/official-cve-feed.json`
+1. `k/website` renders the JSON blob as an HTML table for viewing the list of
+   fixed CVEs from a browser at this location:
+   `https://kubernetes.io/docs/reference/issues-security/official-cve-feed`
+   and linked from this
+   page: `https://kubernetes.io/docs/reference/issues-security/security/`
+
+*Notes*:
+
+- A GCS bucket needs to be created. Example PR for this looks
+  like [this](https://github.com/kubernetes/k8s.io/pull/2570/files)
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
```

Comment:
Update to use the updated Test Plan section
```suggestion
<!--
<!--
**Note:** *Not required until targeted at a release.*
The goal is to ensure that we don't accept enhancements with inadequate testing.

All code is expected to have adequate tests (eventually with coverage
expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
when drafting this test plan.

[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
-->

[ ] I/we understand the owners of the involved components may require updates to
existing tests to make this code solid enough prior to committing the changes necessary
to implement this enhancement.

##### Prerequisite testing updates

<!--
Based on reviewers feedback describe what additional tests need to be added prior
implementing this enhancement to ensure the enhancements have also solid foundations.
-->

##### Unit tests

<!--
In principle every added code should have complete unit test coverage, so providing
the exact set of tests will not bring additional value.
However, if complete unit test coverage is not possible, explain the reason of it
together with explanation why this is acceptable.
-->

<!--
Additionally, for Alpha try to enumerate the core package you will be touching
to implement this enhancement and provide the current unit coverage for those
in the form of:
- <package>: <date> - <current test coverage>
The data can be easily read from:
https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit

This can inform certain test coverage improvements that we want to do before
extending the production code to implement this enhancement.
-->

- `<package>`: `<date>` - `<test coverage>`

##### Integration tests

<!--
This question should be filled when targeting a release.
For Alpha, describe what tests will be added to ensure proper quality of the enhancement.

For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
https://storage.googleapis.com/k8s-triage/index.html
-->

- <test>: <link to test coverage>

##### e2e tests

<!--
This question should be filled when targeting a release.
For Alpha, describe what tests will be added to ensure proper quality of the enhancement.

For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
https://storage.googleapis.com/k8s-triage/index.html

We expect no non-infra related flakes in the last month as a GA graduation criteria.
-->

- <test>: <link to test coverage>
-->
```